### PR TITLE
Moar filter refactoring

### PR DESCRIPTION
**Pull request by dnahodil**
Some tidy-up in LayerController from Filters work

### CODE REVIEW

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy [hunk 1]
```diff
@@ -25,6 +25,12 @@ class LayerController {
 
     static allowedMethods = [save: "POST", update: "POST", delete: "POST"]
 
+    static final HTTP_200_OK = 200
```
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: Perhaps extract this to a more common file? I bet we can use it in many more places...

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy [hunk 2]
```diff
@@ -642,45 +647,52 @@ class LayerController {
     }
 
     def getStylesAsJSON = {
-        if (hostVerifier.allowedHost(params.server)) {
-            def server = params.server
-            def layer = params.layer
-            def serverObject = _getServerClass(params.serverType)
 
-            render text: serverObject.getStyles(server, layer) as JSON
+        def (server, layer, serverType) = parseParams(params)
```
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: That's slick. I didn't know grails does that for you.

Comment (dnahodil) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: Yeah, it's a neat trick.

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy (hunk 1 above)
Comment (dnahodil) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: I'll have a look at expanding this to our other controllers

**Comment by dnahodil** [comment]
I've updated all our controllers to use the HTTP status constants.

I changed all the controllers despite the fact we'll lose a lot of them to no_db changes. It was just easier doing them all.

**Comment by danfruehauf** [comment]
Yep, that looks good.

**Pull request merged**